prov/hook/trace/src/hook_trace.c

```diff
@@ -48,6 +46,99 @@
 	TRACE_EP(ret, ep, count ? iov[0].iov_base : NULL, \
 		 ofi_total_iov_len(iov, count), addr, tagkey, ignoredata, context)
 
+typedef void (*trace_cq_entry_fn)(const struct fi_provider *prov,
+				const char *func, int line,
+				int count, const void *buf, uint64_t data);
+
+static void
+dummy(const struct fi_provider *prov, const char *func, int line,
+		int count, const void *buf, uint64_t data) { }
+
+static void
+trace_cq_context_entry(const struct fi_provider *prov, const char *func, int line,
+				int count, const void *buf, uint64_t data)
+{
+	struct fi_cq_entry *entry = (struct fi_cq_entry *)buf;
+	for (int i=0; i<count; i++) {
```

Comment:
nit: Please add blank line after variable declaration and rest of code.  I *think* we can drop the case if entry is declared as const.  This comment applies to all functions.

second nit: Please add spaces before & after the math operations in the for loop.  e.g. ```for (i = 0; i < count; i++)```